File: human_claude.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        vector<int> a(n);
        for (int i = 0; i < n; i++) cin >> a[i];

        vector<int> nextGreater(n, n);
        stack<int> st;

        for (int i = 0; i < n; i++) {
            while (!st.empty() && a[st.top()] < a[i]) {
                nextGreater[st.top()] = i;
                st.pop();
            }
            st.push(i);
        }

        for (int i = 0; i < n; i++) {
            int ans;
            if (nextGreater[i] != n) {
                ans = nextGreater[i] - i;
            } else {
                ans = n - 1 - i;
            }
            cout << ans;
            if (i != n - 1) cout << " ";
        }
        cout << "\n";
    }

    return 0;
}

File: human_claude.py
import sys

def solve():
    input_data = sys.stdin.read().split()
    pos = 0

    t = int(input_data[pos]); pos += 1
    results = []

    for _ in range(t):
        n = int(input_data[pos]); pos += 1
        a = input_data[pos:pos + n]
        pos += n
        a = [int(x) for x in a]

        next_greater = [n] * n
        stack = []

        for i in range(n):
            while stack and a[stack[-1]] < a[i]:
                j = stack.pop()
                next_greater[j] = i
            stack.append(i)

        ans = []
        for i in range(n):
            if next_greater[i] != n:
                ans.append(next_greater[i] - i)
            else:
                ans.append(n - 1 - i)

        results.append(" ".join(map(str, ans)))

    print("\n".join(results))

solve()